Resources/Prototypes/_NF/Entities/Objects/Weapons/Guns/turrets.yml

```
@@ -0,0 +1,313 @@
+    state: packed
+  - type: Item
+    size: Ginormous
+    shape:
+    - 0,0,3,3
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+
+- type: constructionGraph # deconstruction graph
+  id: WeaponTurretLaserNanoTrasenPackingGraph
+  start: WeaponTurretLaserNanoTrasenPacked
+  graph:
+  - node: WeaponTurretLaserNanoTrasenPacked
+    entity: WeaponTurretLaserNanoTrasenDeployed
+    edges:
+    - to: WeaponTurretLaserNanoTrasenPacked
+      completed:
+      - !type:SpawnPrototype
+            prototype: WeaponTurretLaserNanoTrasenPacked
+            amount: 1
+      - !type:EmptyAllContainers
+      - !type:DeleteEntity {}
+      steps:
+        - tool: Screwing
+          doAfter: 14
+
+- type: entity
+  parent: BaseWeaponTurretEnergyNF
+  id: WeaponTurretLaserSyndicateNF
+  name: laser turret
+  suffix: Frontier, Syndicate
+  components:
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+  - type: Gun
+    fireRate: 2
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/laser.ogg
+      params:
+        variation: 0.125
+  - type: HitscanBatteryAmmoProvider
+    proto: RedLightLaser
+    fireCost: 10
+  - type: NpcFactionMember
+    factions:
+    - Syndicate
+
+- type: entity
+  parent: WeaponTurretLaserSyndicateNF
+  id: WeaponTurretLaserHostileNF
+  name: laser turret
+  suffix: Frontier, Hostile
+  components:
+  - type: NpcFactionMember
+    factions:
+    - SimpleHostile
+
+## Magazine fed, deployable NanoTrasen turret
+- type: entity # deployed turret
+  parent:
+  - BaseWeaponTurretMagazineFed
+  id: WeaponTurretAsmgtNtDeployed
+  name: ASMGT
+  description: |-
+   An Autonomous Sentry Machine Gun Turret (ASMGT) is a magazine fed ballistic turret compatible with .20, .25, .30 and .35 cartriges. Use screwdriver to disassemble.
+  suffix: NT
+  components:
+  - type: NPCImprintingOnSpawnBehaviour
+    whitelist:
+      components:
+      - HumanoidAppearance
+  - type: NpcFactionMember
+    factions:
+    - NanoTrasen
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/magazinefed.rsi
+    drawdepth: Mobs
+    layers:
+    - state: base
+      map: ["enum.GunVisualLayers.Base"]
+    - state: mag-0
+      map: ["enum.GunVisualLayers.Mag"]
+  - type: Appearance
+  - type: MagazineVisuals
+    magState: mag
+    steps: 11
+    zeroVisible: true
+  - type: ItemSlots
+    slots:
+      gun_magazine:
+        name: Magazine
+        startingItem: null
+        insertSound: /Audio/Weapons/Guns/MagIn/smg_magin.ogg
+        ejectSound: /Audio/Weapons/Guns/MagOut/smg_magout.ogg
+        priority: 2
+        whitelist:
+          tags:
+          - MagazineAsmgt
+  - type: Construction
+    deconstructionTarget: null
+    graph: WeaponTurretAsmgtNtPackingGraph
+    node: WeaponTurretAsmgtNtPacked
+
+- type: entity # folded turret
+  parent: BaseItem
+  id: WeaponTurretAsmgtNtPacked
+  name: packed ASMGT
+  description: |-
+   A disassembled and packed Autonomous Sentry Machine Gun Turret (ASMGT).
```

In before somebody guns down Pun Pun with a turret.
```suggestion
   A disassembled and packed Autonomous Sentry Machine Gun Turret (ASMGT). Assigns nearby humanoids as friendly when deployed. All others will be treated as hostile!
```

---

> A disassembled and packed Autonomous Sentry Machine Gun Turret (ASMGT). WARNING! Unpacked turret will be hostile towards anyone who wasn't in close proximity (withing 3 m) during unpacking!

Hm? How about this one?

---

A disassembled and packed Autonomous Sentry Machine Gun Turret (ASMGT). WARNING! Unpacked turret will be hostile towards anyone who wasn't in close proximity (within 3 m) after unpacking!

---

Sure, sounds good.